azurerm/helpers/kubernetes/kube_config.go

```diff
@@ -48,6 +68,16 @@ type KubeConfig struct {
 	Preferences    map[string]interface{} `yaml:"preferences,omitempty"`
 }
 
+type KubeConfigAAD struct {
```

Comment:
Since `Users` field in this struct is the only different one compared to `KubeConfig`, can we build a new base struct, 'KubeConfigBase' and use composition for this struct?

Reply:
Refactored for simplicity.